picard/ui/itemviews.py

```diff
@@ -404,6 +404,83 @@ def __str__(self):
         return f"{name}'s header"
 
 
+def _alternative_versions(album):
+    config = get_config()
+    versions = album.release_group.versions
+
+    album_tracks_count = album.get_num_total_files() or len(album.tracks)
+    preferred_countries = set(config.setting['preferred_release_countries'])
+    preferred_formats = set(config.setting['preferred_release_formats'])
+    ORDER_BEFORE, ORDER_AFTER = 0, 1
+
+    alternatives = []
+    for version in versions:
+        trackmatch = countrymatch = formatmatch = ORDER_BEFORE
+        if version['totaltracks'] != album_tracks_count:
+            trackmatch = ORDER_AFTER
+        if preferred_countries:
+            countries = set(version['countries'])
+            if not countries or not countries.intersection(preferred_countries):
+                countrymatch = ORDER_AFTER
+        if preferred_formats:
+            formats = set(version['formats'])
+            if not formats or not formats.intersection(preferred_formats):
+                formatmatch = ORDER_AFTER
+        group = (trackmatch, countrymatch, formatmatch)
+        # order by group, name, and id on push
+        heappush(alternatives, (group, version['name'], version['id'], version['extra']))
+
+    while alternatives:
+        yield heappop(alternatives)
+
+
+def _build_other_versions_actions(releases_menu, album, alternative_versions):
+    heading = QtGui.QAction(album.release_group.version_headings, parent=releases_menu)
+    heading.setDisabled(True)
+    font = heading.font()
+    font.setBold(True)
+    heading.setFont(font)
+    yield heading
+
+    count = DEFAULT_ALT_VERSIONS_MENU_LIMIT
+    prev_group = None
+    for group, action_text, release_id, extra in alternative_versions:
+        count -= 1
+        if count < 0:
+            action = QtGui.QAction("…", parent=releases_menu)
+            action.setCheckable(False)
+            action.setDisabled(True)
+            action.setToolTip(_("Too many entries, use More Details and/or adjust query limit in Advanced Options"))
```

Comment:
Yes, that's tricky because we have 2 limits in game here, one for the size of the displayed list, one for the size of result list.
The message isn't great, it could be improved by knowing what's the cause of the limit:
- we need to get the total number of results (for example T=120)
- we need to compare that to the number of retrieved results (for example R=60)
- we need  to compare that to the number of displayed items (for example D=30)

We have various cases:
- display limit vs retrieved results limit
- retrieved results limit vs total number of possible results

If D >= R >= T: all possible results are displayed
If D >= R < T: only partial results, but all are displayed 
If D < R < T: partial results, and not all are displayed